### PR TITLE
Fix doxygen warnings

### PR DESCRIPTION
**Pull request by utzig**
This fixes all Doxygen warnings for Doxygen 1.8.13 (the version used by the CI), there are a few warnings left for Doxygen 1.8.15 (and maybe later versions as well).

JIRA: NCSDK-4971

### CODE REVIEW

File: include/bluetooth/mesh/light_ctrl_srv.h [hunk 1]
```diff
@@ -200,7 +200,7 @@ int bt_mesh_light_ctrl_srv_on(struct bt_mesh_light_ctrl_srv *srv);
  *  state). Calling this function temporarily disables occupancy sensor
  *  triggering (referred to as "manual mode" in the documentation). The server
  *  will remain in manual mode until the manual mode timer expires, see
- *  @ref CONFIG_BT_MESH_LIGHT_CTRL_SRV_TIME_MANUAL.
+ *  @em CONFIG_BT_MESH_LIGHT_CTRL_SRV_TIME_MANUAL.
```
Comment (joerchan) [inline include/bluetooth/mesh/light_ctrl_srv.h:203]: Would this work?
:option:`CONFIG_BT_MESH_LIGHT_CTRL_SRV_TIME_MANUAL`

**Comment by joerchan** [comment]
@ru-fu @utzig Can't we get the ref to zephyr kernel defines to work properly? We now only have documentation for the APIs in NRF, but not for the zephyr kernel APIs. The application is supposed to be able to use both. Can't the zephyr API documentation be included in our build? It would be a much nicer user-experience IMO.

File: include/bluetooth/mesh/light_ctrl_srv.h (hunk 1 above)
Comment (utzig) [reply include/bluetooth/mesh/light_ctrl_srv.h:203]: I'm afraid not, this is syntax used with Sphinx/Rst, not doxygen.

Comment (trond-snekvik) [reply include/bluetooth/mesh/light_ctrl_srv.h:203]: We could add an alias to fix this:
```
"option{1}=<a href=\"@KCONFIG_OUTPUT@/\1.html\">\1</a>"
```
then use it in the headers like this:
```
@option{CONFIG_2ND_LEVEL_INTERRUPTS}
```

Comment (utzig) [reply include/bluetooth/mesh/light_ctrl_srv.h:203]: This is quite clever, and it did work on my machine, but unfortunately it does use absolute file paths not URLs, which would not allow it to work on the server when published. For intersphinx we generate the appropriate relative paths with some python calls so something similar can be tried here...

File: include/bluetooth/gatt_pool.h [hunk 2]
```diff
@@ -106,7 +106,7 @@ extern "C" {
  *
  *  @param _gp GATT service object with dynamic attribute allocation.
  *  @param _ccc CCC descriptor configuration.
- *  @param _cb CCC descriptor callback.
+ *  @param _ccc_changed CCC configuration changed callback.
```
Comment (grochu) [inline include/bluetooth/gatt_pool.h:109]: CCC is "Client Characteristic Configuration" so I would change to : "CCC value changed callback" or "CCC changed callback"

Comment (utzig) [reply include/bluetooth/gatt_pool.h]: Done.

**Pull request merged**